Failure on nightly

Running the tests on nightly gives
```julia
separatelabel: Error During Test at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:18
  Test threw exception AssertionError("isexpr(ex.args[2], :(:))")
  Expression: MOIU.separatelabel($(Expr(:quote, :(variables:x)))) == (:variables, :x)
  AssertionError: isexpr(ex.args[2], :(:))
  Stacktrace:
   [1] separatelabel(::Expr) at /home/blegat/.julia/v0.7/MathOptInterface/src/Utilities/parser.jl:175
   [2] macro expansion at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:18 [inlined]
   [3] macro expansion at /home/blegat/git/julia/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
   [4] top-level scope at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:18
separatelabel: Error During Test at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:19
  Test threw exception AssertionError("isexpr(ex.args[1], :(:))")
  Expression: MOIU.separatelabel($(Expr(:quote, :((variables:x, y))))) == (:variables, $(Expr(:quote, :((x, y)))))
  AssertionError: isexpr(ex.args[1], :(:))
  Stacktrace:
   [1] separatelabel(::Expr) at /home/blegat/.julia/v0.7/MathOptInterface/src/Utilities/parser.jl:169
   [2] macro expansion at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:19 [inlined]
   [3] macro expansion at /home/blegat/git/julia/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
   [4] top-level scope at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:18
separatelabel: Error During Test at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:20
  Test threw exception AssertionError("isexpr(ex.args[2], :(:))")
  Expression: MOIU.separatelabel($(Expr(:quote, :(minobjective:x + y)))) == (:minobjective, $(Expr(:quote, :(x + y))))
  AssertionError: isexpr(ex.args[2], :(:))
  Stacktrace:
   [1] separatelabel(::Expr) at /home/blegat/.julia/v0.7/MathOptInterface/src/Utilities/parser.jl:175
   [2] macro expansion at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:20 [inlined]
   [3] macro expansion at /home/blegat/git/julia/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
   [4] top-level scope at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:18
separatelabel: Error During Test at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:21
  Test threw exception AssertionError("isexpr(ex.args[2], :(:))")
  Expression: MOIU.separatelabel($(Expr(:quote, :(con1:2x <= 1)))) == (:con1, $(Expr(:quote, :(2x <= 1))))
  AssertionError: isexpr(ex.args[2], :(:))
  Stacktrace:
   [1] separatelabel(::Expr) at /home/blegat/.julia/v0.7/MathOptInterface/src/Utilities/parser.jl:175
   [2] macro expansion at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:21 [inlined]
   [3] macro expansion at /home/blegat/git/julia/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
   [4] top-level scope at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:18
separatelabel: Error During Test at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:22
  Test threw exception AssertionError("isexpr(ex.args[2], :(:))")
  Expression: MOIU.separatelabel($(Expr(:quote, :(con1:[x, y] in S)))) == (:con1, $(Expr(:quote, :([x, y] in S))))
  AssertionError: isexpr(ex.args[2], :(:))
  Stacktrace:
   [1] separatelabel(::Expr) at /home/blegat/.julia/v0.7/MathOptInterface/src/Utilities/parser.jl:175
   [2] macro expansion at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:22 [inlined]
   [3] macro expansion at /home/blegat/git/julia/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
   [4] top-level scope at /home/blegat/.julia/v0.6/MathOptInterface/test/parser.jl:18
Test Summary: | Error  Total
separatelabel |     5      5
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,17 +5,11 @@ os:
 #  - osx
 julia:
   - 0.6
-#  - nightly
+  - 0.7
 notifications:
   email: false
 git:
   depth: 99999999
-
-## uncomment the following lines to allow failures on nightly julia
-## (tests will run but not make your overall status red)
-matrix:
-  allow_failures:
-  - julia: nightly
 
 # Integration with JuMP-dev gitter channel
 notifications:

--- a/src/Utilities/parser.jl
+++ b/src/Utilities/parser.jl
@@ -1,22 +1,19 @@
 using Base.Meta: isexpr
 
-
-"""
-A parser for a simple human-readable of an MOI model.
-This should be thought of as a compact way to write small models
-for tests, and not an exchange format.
-
-variables: x, y, z
-minobjective: 2x + 3y
-con1: x + y <= 1
-con2: [x,y] in Set
-
-special labels: variables, minobjective, maxobjective
-everything else denotes a constraint with a name
-all constraints must be named
-"x - y" does NOT currently parse, needs to be written as "x + -1.0*y"
-"x^2" does NOT currently parse, needs to be written as "x*x"
-"""
+# A parser for a simple human-readable of an MOI model.
+# This should be thought of as a compact way to write small models
+# for tests, and not an exchange format.
+#
+# variables: x, y, z
+# minobjective: 2x + 3y
+# con1: x + y <= 1
+# con2: [x,y] in Set
+#
+# special labels: variables, minobjective, maxobjective
+# everything else denotes a constraint with a name
+# all constraints must be named
+# "x - y" does NOT currently parse, needs to be written as "x + -1.0*y"
+# "x^2" does NOT currently parse, needs to be written as "x*x"
 
 struct ParsedScalarAffineTerm
     coefficient::Float64
@@ -151,23 +148,45 @@ function parsefunction(ex)
 end
 
 # see tests for examples
-function separatelabel(ex)
-    if isexpr(ex, :(:))
-        return ex.args[1], ex.args[2]
-    elseif isexpr(ex, :tuple)
-        ex = copy(ex)
-        @assert isexpr(ex.args[1], :(:))
-        label = ex.args[1].args[1]
-        ex.args[1] = ex.args[1].args[2]
-        return label, ex
-    elseif isexpr(ex, :call)
-        ex = copy(ex)
-        @assert isexpr(ex.args[2], :(:))
-        label = ex.args[2].args[1]
-        ex.args[2] = ex.args[2].args[2]
-        return label, ex
-    else
-        error("Unrecognized expression $ex")
+if VERSION > v"0.7-"
+    function separatelabel(ex)
+        if isexpr(ex, :call) && ex.args[1] == :(:)
+            return ex.args[2], ex.args[3]
+        elseif isexpr(ex, :tuple)
+            ex = copy(ex)
+            @assert isexpr(ex.args[1], :call) && ex.args[1].args[1] == :(:)
+            label = ex.args[1].args[2]
+            ex.args[1] = ex.args[1].args[3]
+            return label, ex
+        elseif isexpr(ex, :call)
+            ex = copy(ex)
+            @assert isexpr(ex.args[2], :call) && ex.args[2].args[1] == :(:)
+            label = ex.args[2].args[2]
+            ex.args[2] = ex.args[2].args[3]
+            return label, ex
+        else
+            error("Unrecognized expression $ex")
+        end
+    end
+else
+    function separatelabel(ex)
+        if isexpr(ex, :(:))
+            return ex.args[1], ex.args[2]
+        elseif isexpr(ex, :tuple)
+            ex = copy(ex)
+            @assert isexpr(ex.args[1], :(:))
+            label = ex.args[1].args[1]
+            ex.args[1] = ex.args[1].args[2]
+            return label, ex
+        elseif isexpr(ex, :call)
+            ex = copy(ex)
+            @assert isexpr(ex.args[2], :(:))
+            label = ex.args[2].args[1]
+            ex.args[2] = ex.args[2].args[2]
+            return label, ex
+        else
+            error("Unrecognized expression $ex")
+        end
     end
 end
 
@@ -185,14 +204,14 @@ parsedtoMOI(model, s::Union{Float64, Int64}) = s
 for typename in [:ParsedScalarAffineTerm,:ParsedScalarAffineFunction,:ParsedVectorAffineTerm,:ParsedVectorAffineFunction,
                  :ParsedScalarQuadraticTerm,:ParsedScalarQuadraticFunction,:ParsedVectorQuadraticTerm,:ParsedVectorQuadraticFunction,
                  :ParsedSingleVariable,:ParsedVectorOfVariables]
-    moiname = parse(replace(string(typename), "Parsed" => "MOI."))
+    moiname = Compat.Meta.parse(replace(string(typename), "Parsed" => "MOI."))
     fields = fieldnames(eval(typename))
     constructor = Expr(:call, moiname, [Expr(:call,:parsedtoMOI,:model,Expr(:.,:f,Base.Meta.quot(field))) for field in fields]...)
     @eval parsedtoMOI(model, f::$typename) = $constructor
 end
 
 function loadfromstring!(model, s)
-    parsedlines = filter(ex -> ex != nothing,parse.(split(s,"\n")))
+    parsedlines = filter(ex -> ex != nothing, Compat.Meta.parse.(split(s,"\n")))
 
     for line in parsedlines
         label, ex = separatelabel(line)
@@ -221,7 +240,7 @@ function loadfromstring!(model, s)
             f = parsedtoMOI(model, parsefunction(ex.args[2]))
             if ex.args[1] == :in
                 # Could be safer here
-                set = eval(MOI, ex.args[3])
+                set = Compat.Core.eval(MOI, ex.args[3])
             elseif ex.args[1] == :<=
                 set = MOI.LessThan(ex.args[3])
             elseif ex.args[1] == :>=

